core/modules/qana/QueryMapping.cc

```diff
@@ -116,6 +116,10 @@ class Mapping : public query::QueryTemplate::EntryMapping {
         //if (!e.isDynamic()) {return newE; }
 
         for(auto i=_tupleDeque.begin(); i != _tupleDeque.end(); ++i) {
+            if (i->param == QueryMapping::SUBCHUNK) {
+                // don't do subchunk replacement here; the worker will take care of it.
+                continue;
+            }
             newE->s = replace(newE->s, i->pat, i->tgt);
```

Comment:
Checking for the subchunk identifier string here works, but IMO this is the main area that still needs more work: I think the more correct fix is to remove the subchunk from `_tupleDeck` entirely. I need to do a couple days' work to understand the classes involved including `QuerySession`, `ChunkSpec`, `ChunkSpecFragmenter`, and design the subchunk string handling out of those classes.